### PR TITLE
feat: add check for bam files

### PR DESCRIPTION
**Pull request by YaroslavTrefilovBraveDevelopers**

### CODE REVIEW

File: src/pages/igv/page.tsx [hunk 1]
```diff
@@ -1,18 +1,19 @@
...
+import { FileMissing } from './ui/file-missing'
 
 const igv = require('igv')
 
-// TODO: when the design is ready, make an error if the file you need is missing
-
-// TODO: add check if we downloaded this sample...
 const IgvPage = observer(
   (): ReactElement => {
+    const [test, setTest] = useState<boolean>(false)
```
Comment (AndreyUstyumenko) [inline src/pages/igv/page.tsx:15]: what means 'test'?

Comment (YaroslavTrefilovBraveDevelopers) [reply src/pages/igv/page.tsx:15]: I forgot to rename, I'll fix it now

File: src/pages/igv/types/ListBucket.inerface.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  _text: string
+}
+
+interface ID {
+  _text: string
+}
+
+interface DisplayName {
+  _text: string
+}
```
Comment (AndreyUstyumenko) [inline src/pages/igv/types/ListBucket.inerface.ts:23]: define a common interface 
interface IText: {
  _text: string
} and replace all duplicates with it.

File: src/pages/igv/serviceProvider/igvServiceProvider.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
+import { xml2json } from 'xml-js'
+
+import { getIgvUrl } from '@core/get-api-url'
+import { IListBucket } from '../types/ListBucket.inerface'
```
Comment (AndreyUstyumenko) [inline src/pages/igv/serviceProvider/igvServiceProvider.ts:4]: typo inerface -> interface

File: src/pages/igv/igv.page.tsx [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+
+    const isCorrectParams = locus && names
+
+    useEffect(() => {
+      igvServiceProvider
+        .getListBucketAsync()
+        .then(listBucket => convertToListObject(listBucket))
+        .then(list => setListObject(list))
+        // eslint-disable-next-line no-console
+        .catch(error => console.error(error))
```
Comment (AndreyUstyumenko) [inline src/pages/igv/igv.page.tsx:34]: define a catch inside service provider

**Review by AndreyUstyumenko — APPROVED** [review]

**Review by evgeniy-chernomortsev — APPROVED** [review]

**Review by Lapk1n — APPROVED** [review]

**Pull request closed without merging**